src/Element/LNURLTip.tsx

```diff
@@ -0,0 +1,301 @@
+      setAmount(undefined);
+      setComment(undefined);
+      setSuccess(undefined);
+    }
+  }, [show, service]);
+
+  const serviceAmounts = useMemo(() => {
+    if (payService) {
+      const min = (payService.minSendable ?? 0) / 1000;
+      const max = (payService.maxSendable ?? 0) / 1000;
+      return amounts.filter((a) => a >= min && a <= max);
+    }
+    return [];
+  }, [payService]);
+
+  const metadata = useMemo(() => {
+    if (payService) {
+      const meta: string[][] = JSON.parse(payService.metadata);
+      const desc = meta.find((a) => a[0] === "text/plain");
+      const image = meta.find((a) => a[0] === "image/png;base64");
+      return {
+        description: desc ? desc[1] : null,
+        image: image ? image[1] : null,
+      };
+    }
+    return null;
+  }, [payService]);
+
+  const selectAmount = (a: number) => {
+    setError(undefined);
+    setInvoice(undefined);
+    setAmount(a);
+  };
+
+  async function fetchJson<T>(url: string) {
+    const rsp = await fetch(url);
+    if (rsp.ok) {
+      const data: T = await rsp.json();
+      console.log(data);
+      setError(undefined);
+      return data;
+    }
+    return null;
+  }
+
+  async function loadService(): Promise<LNURLService | null> {
+    if (service) {
+      const isServiceUrl = service.toLowerCase().startsWith("lnurl");
+      if (isServiceUrl) {
+        const serviceUrl = bech32ToText(service);
+        return await fetchJson(serviceUrl);
+      } else {
+        const ns = service.split("@");
+        return await fetchJson(`https://${ns[1]}/.well-known/lnurlp/${ns[0]}`);
+      }
+    }
+    return null;
+  }
+
+  async function loadInvoice() {
+    if (!amount || !payService) return null;
+    let url = "";
+    const amountParam = `amount=${Math.floor(amount * 1000)}`;
+    const commentParam = comment
+      ? `&comment=${encodeURIComponent(comment)}`
+      : "";
+    if (payService.nostrPubkey && author) {
+      const ev = await publisher.zap(author, note, comment);
+      const nostrParam =
+        ev && `&nostr=${encodeURIComponent(JSON.stringify(ev.ToObject()))}`;
+      url = `${payService.callback}?${amountParam}${commentParam}${nostrParam}`;
+    } else {
+      url = `${payService.callback}?${amountParam}${commentParam}`;
+    }
+    try {
+      const rsp = await fetch(url);
+      if (rsp.ok) {
+        const data = await rsp.json();
+        console.log(data);
+        if (data.status === "ERROR") {
+          setError(data.reason);
+        } else {
+          setInvoice(data);
+          setError("");
+          payWebLNIfEnabled(data);
+        }
+      } else {
+        setError("Failed to load invoice");
+      }
+    } catch (e) {
+      setError("Failed to load invoice");
+    }
+  }
+
+  function custom() {
+    const min = (payService?.minSendable ?? 0) / 1000;
+    const max = (payService?.maxSendable ?? 21_000_000_000) / 1000;
+    return (
+      <div className="flex mb10">
+        <input
```

Comment:
We can remove this file and LNURLTip.css since they have been converted to SendSats component